src/gui/qgsmaptoolidentify.cpp

```diff
@@ -374,6 +374,25 @@ void QgsMapToolIdentify::closestVertexAttributes( const QgsAbstractGeometry &geo
   }
 }
 
+void QgsMapToolIdentify::closestPointAttributes( const QgsAbstractGeometry &geometry, const QgsPointXY &layerPoint, QMap<QString, QString> &derivedAttributes )
+{
+  QgsPoint closestPoint = QgsGeometryUtils::closestPoint( geometry, QgsPoint( layerPoint.x(), layerPoint.y() ) );
```

Comment:
Should we have a `QgsPoint::toPointXY` method?

Reply:
Actually there's already `QgsPointXY( const QgsPoint &point )`, I'll just use that

Reply:
Maybe we just want to remove the explicit part there for the direction QgsPointXY -> QgsPoint (and leave it for the other one)

Reply:
Implicit conversions scare me :fearful:

Reply:
Just look at it as upcasting ;)

```
QgsMapLayer* l = myVectorLayer; // ok
QgsVectorLayer* l = myMapLayer; // not gonna be that easy
```